src/lib/middleware/services/sys_info_service.cpp

```diff
@@ -62,8 +63,6 @@ static bool sys_info_service_handler(size_t service_strlen, const char *service,
                                    uint8_t *reply_data) {
   (void)(req_data);
   bool rval = false;
-  size_t max_data_len = buffer_len;
-  buffer_len = 0;
```

Comment:
I had to read carefully around to understand that the message cpp file handles checking the buffer size. If the caller of this function passes a buffer that's too small, then this function will return false because encode will return CborErrorOutOfMemory. Yeah?

Reply:
Yes, exactly! Can add a comment for clarity